src/App.js

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect } from 'react';
+import './App.css';
+import { Routes, Route } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import Navbar from './components/Navbar';
+import Rocket from './components/Rocket';
+import Missions from './pages/Missions';
+import MyProfile from './components/MyProfile';
+import Dragon from './components/Dragon';
+import { getMissions } from './redux/mission/missionSlice';
+import { fetchRockets } from './redux/Rocket/rocketSlice';
+import { getDragon } from './redux/dragon/dragonSlice';
+
+function App() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getMissions());
+    dispatch(fetchRockets());
+    dispatch(getDragon());
+  }, [dispatch]);
```

Comment:
- [ ] You did a great job with most of the functionalities. :+1: Let's work on some more details to achieve the project requirements. According to the requirements, we are supposed to fetch data only when the components are rendered. Kindly make sure the corresponding data is fetched when the components are rendered. Let's move each fetching logic into its respective component. ✍️

**Project requirements**
![Screenshot from 2023-03-24 17-29-50](https://user-images.githubusercontent.com/17243325/227554668-3fbb9266-695d-4ebd-bf25-31cf3c1a524f.png)

Reply:
Thanks for the suggestion @amejid 👍

Reply:
Fixed.